src/edu/hillel/lesson17/realization1/PetrolStation.java

```diff
@@ -0,0 +1,59 @@
+package edu.hillel.lesson17.realization1;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class PetrolStation {
+    private static int amount;
+    private static final int THREAD_POOL_SIZE = 2;
+    private static final int MIN_DELAY = 3000;
+    private static final int MAX_DELAY = 10000;
+    private static final int MIN_REFILL_AMOUNT = 10;
+    private static final int MAX_REFILL_AMOUNT = 50;
+
+    private static final int MINIMAL_FUEL_AMOUNT = MAX_REFILL_AMOUNT * (THREAD_POOL_SIZE + 1);
+
+    public static void main(String[] args) {
+
+        ExecutorService executorService = new ThreadPoolExecutor(THREAD_POOL_SIZE,
+                THREAD_POOL_SIZE,
+                0L,
+                TimeUnit.MILLISECONDS,
+                new ArrayBlockingQueue<>(THREAD_POOL_SIZE),
+                new ThreadPoolExecutor.CallerRunsPolicy());
+
+        amount = 1000;
+
+        while (amount > MINIMAL_FUEL_AMOUNT) {
```

Comment:
What if amount == MINIMAL_FUEL_AMOUNT and not only > ?

Reply:
Yes - ok, I add =. Agree